File: obj-c/28_usov_ch_37/TestProject/TestProject/ViewController.m
//
//  ViewController.m
//  TestProject
//
//  Created by Виталий Цыганенко on 08.04.2023.
//

#import "ViewController.h"

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UITextField* textField;
@property (strong, nonatomic) IBOutlet UILabel* label;

@property (strong, nonatomic) IBOutlet UILabel *myLabel;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

-(IBAction) onUnwindAction: (UIStoryboardSegue*) unwindSegue {
    NSLog(@"ViewController unwind action");
}

-(IBAction) textToLog {
    NSLog(@"%@", self.textField.text);
}

-(IBAction) textToLabel {
    self.label.text = self.textField.text;
}

- (IBAction)onTapGesture:(id)sender {
    [self.textField resignFirstResponder];
}

- (IBAction)changeLabelText:(UIButton *)sender {
    if(sender.titleLabel && sender.titleLabel.text) {
        NSString* buttonText = sender.titleLabel.text;
        [self.myLabel setText:buttonText];
        NSLog(@"%@ button was pressed", buttonText);
    }
}


@end

File: swift/01_usov_ch_37/TestProject/TestProject/ViewController.swift
//
//  ViewController.swift
//  TestProject
//
//  Created by Виталий Цыганенко on 10.04.2023.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet var textField: UITextField!
    @IBOutlet var label: UILabel!

    @IBOutlet var myLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func onUnwindAction(unwindSegue: UIStoryboardSegue) {
        print("ViewController::onUnwindAction")
    }
    
    @IBAction func textToLog() {
        print("ViewController: ", textField.text!)
    }
    
    @IBAction func textToLabel() {
        label.text = textField.text
    }
    
    @IBAction func onTapGesture() {
        textField.resignFirstResponder()
    }

    @IBAction func changeLabelText(_ sender: UIButton) {
        if let buttonText = sender.titleLabel!.text {
            self.myLabel.text = buttonText
            print("\(buttonText) was pressed")
        }
    }
    
}

